### PR TITLE
Doc/user apis lld

### PR DESCRIPTION
**Pull request by DrishikaPatil01**

### CODE REVIEW

**Comment by HarshitCd** [comment]
Create a `doc/` directory and put all the markdown files there.

File: api/ForgotPassword.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+## LLD
+![Forgot Password Level Diagram](./assets/ForgotPasswordLld.png)
+
+
+## Request
+
+### Path
+| **Field** | **Value**                             |
+|-----------|-----------                            |
+| Base Url  | http://localhost:8080/password-manager|
```
Comment (HarshitCd) [inline api/ForgotPassword.md]: is there a reason for having `http://localhost:8080/password-manager` as the base url?
Can't we just keep it as `http://localhost:8080/`?

File: api/ForgotPassword.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+| Path      |    /user/password/set                 |
```
Comment (HarshitCd) [inline api/ForgotPassword.md]: Why is the path `/user/password/set` rather than `/forgot-password`?

File: api/ResetPassword.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+## Request
+
+### Path
+| **Field** | **Value**                               |
+|-----------|-----------                              |
+| Base Url  | http://localhost:8080/password-manager  |
+| Path      |    /user/password/reset                 |
+| Headers   | Content-Type: application/json          |
+
```
Comment (HarshitCd) [inline api/ResetPassword.md]: Don't we need some sort of authentication field to validate the user that is logged in can only reset his/her password?

File: api/Login.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+## Response
+
+### Response Body
+| **Field**          | **Description**                    |
+|--------------------|------------------------------------|
+| status             | status of login  - SUCCESS/FAILED  |
+| userId             |    user's Id assigned by service   |
+| email              |    user's email                    |
+| error.Code         |                                    |
+| error.Description  |                                    |
```
Comment (HarshitCd) [inline api/Login.md]: Will the api return a token back that can be used to validate any request henceforth is done by a valid user?

Comment (HarshitCd) [reply api/Login.md]: If not how are we validating the user authenticity when the make further request when they are logged in?

File: api/ForgotPassword.md (hunk 1 above)
Comment (DrishikaPatil01) [reply api/ForgotPassword.md]: Since we dont have a host name this seemed better

File: api/ForgotPassword.md (hunk 2 above)
Comment (DrishikaPatil01) [reply api/ForgotPassword.md]: Cause we're setting the password

File: api/ResetPassword.md (hunk 3 above)
Comment (DrishikaPatil01) [reply api/ResetPassword.md]: This should be handled in client side, backend doesn't need to handle this

**Pull request closed without merging**